huaweicloud/services/cpts/resource_huaweicloud_cpts_project.go

```diff
@@ -0,0 +1,178 @@
+package cpts
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/cpts/v1/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceProject() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceProjectCreate,
+		UpdateContext: resourceProjectUpdate,
+		DeleteContext: resourceProjectDelete,
+		ReadContext:   resourceProjectRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 42),
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 50),
+			},
+
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceProjectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcCptsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CPTS v1 client: %s", err)
+	}
+
+	createOpts := &model.CreateProjectRequest{
+		Body: &model.CreateProjectRequestBody{
+			Name:        d.Get("name").(string),
+			Description: utils.String(d.Get("description").(string)),
+		},
+	}
+	response, err := client.CreateProject(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating CPTS project: %s", err)
+	}
+
+	d.SetId(fmt.Sprint(*response.ProjectId))
```

Comment:
1. should check whether the field is a nil pointer;
2. suggest to use strconv.Itoa

Reply:
done